src/Form/DataField/Integer.hpp

```diff
@@ -46,8 +46,9 @@ class DataFieldInteger final : public NumberDataField
   DataFieldInteger(const TCHAR *edit_format, const TCHAR *display_format,
                    int _min, int _max, int _value, int _step,
                    DataFieldListener *listener=nullptr) noexcept
-    :NumberDataField(Type::INTEGER, true, edit_format, display_format, listener),
-     value(_value), min(_min), max(_max), step(_step) {}
+    :NumberDataField(Type::INTEGER, ((_max-_min)/_step) < 500, edit_format,
+    display_format, listener), value(_value), min(_min), max(_max),
```

Comment:
Cherry-picked and fixed coding style violations.
82402aee85b1601b354f2f8b6521b28ead964227